app/models/asset.rb

```diff
@@ -74,7 +74,7 @@ def self.search(
                              .pluck(:id)
 
     assets_in_inventories = Asset.joins(repository_cell: { repository_column: :repository })
-                                 .where(repositories: { team: teams })
+                                 .where(repositories: { team: teams.filter { |t| t.permission_granted?(user, RepositoryPermissions::READ) } })
```

Comment:
Why we can't use a subquery here?

Reply:
And actually shouldn't we simply check RepositoryPermissions::READ on repositories?